src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -90,7 +89,7 @@ private AssemblyNameParts Parse()
             if (token != Token.String)
                 ThrowInvalidAssemblyName();
 
-            if (name == string.Empty || name.IndexOfAny(s_illegalCharactersInSimpleName) != -1)
+            if (name == string.Empty || name.AsSpan().IndexOfAny('/', '\\', ':') != -1)
```

Comment:
Nit: changing the first part to `string.IsNullOrEmpty(name)` would offer additional defense against somebody accidentally violating the nullability annotation. The previous code would've thrown if _name_ somehow ended up null, but the new code will not.